graylog2-server/src/main/java/org/graylog2/shared/security/ShiroSecurityContext.java

```diff
@@ -49,18 +51,22 @@ public ShiroSecurityContext(Subject subject, AuthenticationToken token, boolean
         this.headers = new MultivaluedHashMap<String, String>(headers);
     }
 
+    @Nullable
     public String getUsername() {
         if (token == null || token.getPrincipal() == null) {
             return null;
         }
         return token.getPrincipal().toString();
     }
 
+    @Nullable
     public String getPassword() {
-        if (token == null || token.getCredentials() == null) {
+        if (token instanceof UsernamePasswordToken) {
```

Comment:
Hmm, this breaks once we implement another `AuthenticationToken` which returns non-null in `#getCredentials()`.

What's the reason for the type check here?

Reply:
@bernd For now, `UsernamePasswordToken` is the only type which has a sensible value for `#getCredentials()`. We can add handling for other tokens if the need for that comes up.

The previous implementation of the `ShiroSecurityContext#getPassword()` method was simply wrong (and returned `char[].toString()` in case of a `UsernamePasswordToken`).